app/routes/models/calculation-bps.js

```diff
@@ -0,0 +1,168 @@
+        { text: toCurrencyString(x.payment), format: 'numeric', classes: 'govuk-body govuk-!-font-weight-bold' }
+      ]
+    )
+    return x
+  })
+
+  return {
+    caption: options.caption,
+    captionClasses: 'govuk-table__caption--m',
+    firstCellIsHeader: true,
+    head: getSummaryHeaderRow(),
+    rows: data.slice(0, 3)
+  }
+}
+
+function getHeaderRow () {
+  return [
+    {
+      text: 'Payment band',
+      classes: 'govuk-!-width-one-half'
+    },
+    {
+      text: '2021',
+      format: 'numeric'
+    },
+    {
+      text: '2022',
+      format: 'numeric'
+    },
+    {
+      text: '2023',
+      format: 'numeric'
+    }
+  ]
+}
+
+function getSummaryHeaderRow () {
+  return [
+    {
+      text: 'Scheme year'
+    },
+    {
+      text: 'Total estimated reductions',
+      format: 'numeric'
+    },
+    {
+      text: 'Estimated payments',
+      format: 'numeric'
+    }
+  ]
+}
+
+function populateData (calculations, options) {
+  const reductionData = calculations.bandResult.map(x => toRow(x, options.property, options.formatType).slice(0, 4))
+  if (options.showOverall) {
+    reductionData.push(populateOverall(calculations, options.property, options.text).flat().slice(0, 4))
+  }
+  return reductionData
+}
+
+function toRow (results, property, formatType) {
+  const data = []
+  data.push({ text: getBandText(results.band) })
+  results.result.map((x) => {
+    data.push(
+      {
+        text: (formatType === 'currency'
+          ? toCurrencyString(x[property])
+          : calculatePercentage(x, property)),
+        format: 'numeric'
+      })
+    return x
+  })
+  return fillGaps(results, data, formatType)
+}
+
+function getBandText (band) {
+  return bands.find(x => x.band === band).text
+}
+
+function calculatePercentage (x, property) {
+  return `${x.payment > 0 ? Math.round(x[property] * 100) : 0}%`
+}
+
+function fillGaps (results, data, formatType) {
+  const checkSchemeYears = results.result.map(x => x.schemeYear)
+  const maxSchemeYear = Math.max(Math, ...schemeYears)
+  const minSchemeYear = Math.min(Math, ...schemeYears)
+
+  const missingData = {
+    text: (formatType === 'currency'
+      ? '£0.00'
+      : '0%'),
+    format: 'numeric'
+  }
+
+  let yearIncrementCount = 0
+
+  for (let i = minSchemeYear; i <= maxSchemeYear; i++) {
+    yearIncrementCount++
+    if (checkSchemeYears.indexOf(i) < 0) {
+      data.splice(yearIncrementCount, 0, missingData)
+    }
+  }
+
+  return data
+}
+
+function populateOverall (calculations, property, text) {
+  const overall = calculations.overallResult.map((x, index) => overallToRow(x, property, index))
+  overall.unshift({ text })
+  return overall
+}
+
+function overallToRow (overallResult, property, index) {
+  const data = []
+  data.push({
+    schemeYear: schemeYears[index],
+    text: toCurrencyString(overallResult[property]),
+    format: 'numeric',
+    classes: 'govuk-body govuk-!-font-weight-bold'
+  })
+  return data
+}
+
+function createBackLink () {
```

Comment:
This function is redundant, hence it can be removed.

Reply:
redundant function and reference removed